relation_engine_server/utils/json_validation.py

```diff
@@ -114,3 +135,32 @@ def _load_json_schema(file):
             return json.load(fd)
 
         raise TypeError('Unknown file type encountered: ' + file)
+
+
+class ExtendedRefResolver(RefResolver):
+
+    def resolve_remote(self, uri):
+
+        scheme = urlsplit(uri).scheme
+        # if there's no scheme, it's a local file, so prefix it with "file://"
+        if scheme == '':
+            uri = 'file://' + uri
```

Comment:
Insert - assume `$ref`s in the form `/path/to/reference.json` are local files, to be accessed using the `file://` protocol

Reply:
So with a relative reference like `my/file.json` it will make the `$ref` into `file://my/file.json`? For those to work, you would have to get the current working directory and insert that into the path. `file://` is always absolute.

If we serve these schemas over http at some point, and they are served with relative paths in the `$ref` without any `$id` being set, then we will have to set some base URIs from the client as the server URL

Reply:
At the moment, these refs are only followable if you give the RefResolver a file path as well as a schema -- if you just use a schema, there's no way for the ref resolver to work out where the file is.

I have written a hack that injects an `$id` field into the schemas to solve the serving over http problem.